heron/api/src/java/com/twitter/heron/api/topology/BaseComponentDeclarer.java

```diff
@@ -64,6 +66,10 @@ public void dump(TopologyAPI.Component.Builder bldr) {
 
     TopologyAPI.Config.Builder cBldr = TopologyAPI.Config.newBuilder();
     for (Map.Entry<String, Object> entry : componentConfiguration.entrySet()) {
+      if (entry.getValue() == null) {
```

Comment:
Check entry.getKey() too since the key can also be null.
https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#put(K,%20V)